src/js/hooks/inboundV2/addItems/useInboundAddItemsActions.js

```diff
@@ -381,64 +352,56 @@ const useInboundAddItemsForm = ({
         }));
 
       if (mismatchedItems.length > 0) {
-        const shouldUpdate = await confirmExpirationDateSave(mismatchedItems);
+        const shouldUpdate = await openModal(
+          { data: mismatchedItems, type: modalWithTableType.EXPIRATION },
+        );
         if (!shouldUpdate) {
           return Promise.reject();
         }
       }
     }
 
-    return updateInventoryItemsAndTransitionToNextStep(updatedValues, lineItems);
+    return updateInventoryItemsAndTransitionToNextStep(lineItems);
   };
 
-  const saveAndTransitionToNextStep = async (values, lineItems) => {
+  const saveAndTransitionToNextStep = async (lineItems) => {
     try {
       spinner.show();
       const resp = await saveRequisitionItemsInCurrentStep(lineItems);
-      const updatedValues = resp
-        ? {
-          ...values,
-          lineItems: resp.data.data.lineItems,
-        }
-        : values;
-
-      handleTransition(updatedValues, lineItems);
+      const updatedLineItems = resp ? resp.data.data.lineItems : [];
+      handleTransition({ updatedLineItems, lineItems });
     } finally {
-      resetFocus();
       spinner.hide();
     }
   };
 
-  const getItemsMapByProductCode = (lineItems) =>
-    lineItems.reduce((acc, item) => {
-      const { productCode } = item.product;
-      return {
-        ...acc,
-        [productCode]: [...(acc[productCode] || []), item],
-      };
-    }, {});
-
-  const checkDuplicatesSaveAndTransitionToNextStep = (formValues, lineItems) => {
-    const itemsMap = getItemsMapByProductCode(lineItems);
-
-    const itemsWithSameCodeFlattened = Object.values(itemsMap)
-      .filter((item) => item.length > 1)
-      .flat();
-    if (Object.values(itemsMap)
-      .some((item) => item.length > 1) && !(formValues.values.origin.locationType.locationTypeCode === 'SUPPLIER'
-      || !formValues.values.hasManageInventory)) {
-      confirmTransition(
-        () => saveAndTransitionToNextStep(formValues, lineItems),
-        itemsWithSameCodeFlattened,
-      );
-    } else {
-      saveAndTransitionToNextStep(formValues, lineItems);
+  const checkDuplicatesSaveAndTransitionToNextStep = async (formValues, lineItems) => {
+    const itemsMap = _.groupBy(
+      lineItems.filter(item => item.product?.productCode),
+      (item) => item.product.productCode,
+    );
+    const duplicateGroups = Object.values(itemsMap).filter((g) => g.length > 1);
+    const hasDuplicates = duplicateGroups.length > 0;
+    const skipConfirm =
+      formValues.values.origin.locationType.locationTypeCode === locationType.SUPPLIER ||
```

Comment:
move `||` to the new line (ands, ors etc should be placed at the beginning of the line)